SampleShared/Samples/TextPanel/TextPanel.cs

```diff
@@ -57,10 +57,17 @@ protected void Update()
         {
             if (m_textProviders != null && m_textProviders.Count > 0)
             {
+                bool shouldAddLineBreak = false;
                 StringBuilder stringBuilder = new StringBuilder();
                 foreach (var textProvider in m_textProviders)
                 {
+                    if (shouldAddLineBreak)
+                    {
+                        stringBuilder.AppendLine();
+                    }
+
                     stringBuilder.Append(textProvider.UpdateText());
```

Comment:
Would using `AppendLine` here work as well? Or does the trailing new line cause formatting problems for the final line?

```suggestion
                    stringBuilder.AppendLine(textProvider.UpdateText());
```

Reply:
We don't want to add a blank line if there's only 1 provider.  as psuedo code, this will be :

```
if (shouldAddLineBreak)
    stringBuilder.AppendLine(textProvider.UpdateText());
else 
    stringBuilder.Append(textProvider.UpdateText());
```